src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separator);
+  const dateComponents = {};
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  fromFormat.forEach((el, index) => {
+    if (el !== separator) {
+      dateComponents[el] = dateParts[index];
+    }
+  });
+
+  if (dateComponents['YY']) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? '20' + dateComponents['YY']
+        : '19' + dateComponents['YY'];
```

Comment:
The condition here assumes that the 'YY' format will always be a number less than 100. If the input is not as expected, this could lead to incorrect results. It would be safer to validate the input before performing this operation.